apps/dashboard/app/models/files.rb

```diff
@@ -0,0 +1,87 @@
+class Files
+
+  def self.raise_if_cant_access_directory_contents(path)
+    # try to actually access directory contents
+    # if an exception is raised, we do not have access
+    path.each_child.first
+  end
+
+  def ls(dirpath)
+    Pathname.new(dirpath).each_child.map do |path|
+      Files.stat(path)
+    end.sort_by { |p| p[:directory] ? 0 : 1 }
+  end
+
+  def self.stat(path)
+    path = Pathname.new(path)
+
+    # path.stat will not work for a symlink and will raise an exception
+    # if the directory or file being pointed at does not exist
+    begin
+      s = path.stat
+    rescue
+      s = path.lstat
+    end
```

Comment:
I cannot replicate the error, though I do see a difference in the mode and so on.  Seems like this would always return `stat` and never rescue to `lstat`.  Maybe it should be `if path.symlink?`?

```text
[jeff 04:58:00 dev()] 🐷 irb
2.7.1 :001 > Pathname.new('shell').stat
 => #<File::Stat dev=0xfd00, ino=4069384, mode=040775, nlink=9, uid=1000, gid=1000, rdev=0x0, size=4096, blksize=4096, blocks=8, atime=2021-04-07 16:20:50.305353909 -0400, mtime=2021-04-07 14:24:36.489150406 -0400, ctime=2021-04-07 14:24:36.489150406 -0400> 
2.7.1 :002 > Pathname.new('shell').lstat
 => #<File::Stat dev=0xfd00, ino=671701, mode=0120777, nlink=1, uid=1000, gid=1000, rdev=0x0, size=28, blksize=4096, blocks=0, atime=2021-04-08 16:58:00.787719412 -0400, mtime=2020-02-06 11:29:06.065785409 -0500, ctime=2020-02-06 11:29:06.065785409 -0500> 
```

Reply:
Will improve symlink reporting in app in future release.